### DISCUSSION TITLE
ERROR: connecting to full node (libp2p): no good addresses

### DISCUSSION THREAD

**Discussion by perezja** [filecoin-project/lotus#9415] [Storage Provider] [answered]
I am having trouble spinning up my miner due to possible connectivity issue with the full node API. 

Lotus miner config

```
[API]
  ListenAddress = "/ip4/0.0.0.0/tcp/6669/http"
  RemoteListenAddress = "192.168.1.32:6669"
[Libp2p]
  ListenAddresses = ["/ip4/0.0.0.0/tcp/6668", "/ip6/::/tcp/6668"]
  AnnounceAddresses = ["/ip4/184.xxx.xxx.xxx/tcp/6668"]
```

Full node config

```
# Lotus daemon config:
[API]
  ListenAddress = "/ip4/192.168.1.31/tcp/6671/http"
[Libp2p]
  ListenAddresses = ["/ip4/0.0.0.0/tcp/6672"]
  AnnounceAddresses = ["/ip4/184.xxx.xxx.xxx/tcp/6672/http"]
```

$FULLNODE_API_INFO

```
eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.TnTfl-9Fr0RGWfFVpYZD0sDMrFFzO6C9sWdXCtdJ_1Q:/ip4/192.168.1.31/tcp/6671/http
```

Running the miner

```
root@athens:/home/debian/lotus-miner# make miner
lotus-miner run
Worker: 22085; Base: 2220274; Target: 2220274 (diff: 0)
State: complete; Current Epoch: 2220274; Todo: 0

Done!
[flexi_logger][ERRCODE::Time] flexi_logger has to work with UTC rather than with local time, caused by IndeterminateOffset                                                                                 
    See https://docs.rs/flexi_logger/latest/flexi_logger/error_info/index.html#time
ERROR: connecting to full node (libp2p): no good addresses
```

Anything immediately jump out as obvious? The miner is obviously communicating with the full node API to get the chain state before outputting the error. So the issue must be with the listen/announce addresses in the full node Libp2p section, whose ports (6671, 6672) are properly forwarded by the gateway to the indicated private IP of the full node (192.168.1.31). Additionally, the miner libp2p port (6672) is forwarded to the miner private IP, so I am lost on the issue (I have had this setup working before on a previous mainnet version earlier in the year).

**Comment by LexLuthr** [comment, marked as answer]
Try to remove `http` from `Libp2p.AnnounceAddresses` from daemon config and restart the daemon. This should allow miner to connect with the daemon over libp2p and add the protected peer.

**Reply by LexLuthr** [reply]
AnnounceAddresses = ["/ip4/184.xxx.xxx.xxx/tcp/6672"] in case it was not very clear

**Reply by perezja** [reply]
Thanks so much! Must have been left on when copy-pasting addresses between api and libp2p sections.
